controller/volume_controller.go

```diff
@@ -625,6 +625,16 @@ func (vc *VolumeController) ReconcileEngineReplicaState(v *longhorn.Volume, e *l
 			}
 		}
 
+		setting, err := vc.getAutoBalancedReplicasSetting(v)
+		if err != nil {
+			return err
+		}
+		if setting != types.ReplicaAutoBalanceDisabled {
+			if err := vc.replenishReplicas(v, e, rs, ""); err != nil {
```

Comment:
I assume this `replenishReplicas` will respect the result of the previous `replenishReplicas` call happing when data locality is enabled.

Reply:
replica auto-balance count only happens when all replicas are deployed(`rCount == usableCount`), this shouldn't conflict with when data locality is enabled which happens when replicas are still replenishing to meet the `v.Spec.NumberOfReplicas`.